R/create_default_parameters.R

```diff
@@ -348,54 +348,103 @@ create_default_fleet <- function(fleets,
+                                          "DlnormDistribution" = create_default_DlnormDistribution(
+                                            value = index_uncertainty,
+                                            input_type = "data",
+                                            data = data
+                                          )
+     )
+     names(index_distribution_default) <- paste0(
+       index_distribution,
+       ".",
+       names(index_distribution_default)
+     )
+
+  }else{
+    q_default <- list(
+      log_q.value = 0,
+      log_q.estimated = FALSE
     )
+
+    index_distribution_default <- NULL
   }
 
-  names(process_default) <- paste0("Fleet.", names(process_default))
+  names(q_default) <- paste0("Fleet.", names(q_default))
 
-  # Create index distribution defaults
-  index_distribution <- fleets[[fleet_name]][["data_distribution"]]["Index"]
+  if ("landings" %in% data_types_present &&
+                   "Landings" %in% distribution_names_for_fleet) {
+    log_Fmort_default <- list(
+      log_Fmort.value = rep(-3, get_n_years(data)),
+      log_Fmort.estimated = TRUE
+    )
 
-  # FIXME: Will this work if both landings and index data are present?
-  index_uncertainty <- get_data(data) |>
-    dplyr::filter(name == fleet_name, type %in% c("landings", "index")) |>
-    dplyr::arrange(dplyr::desc(type)) |>
-    dplyr::pull(uncertainty)
+    landings_distribution <- fleets[[fleet_name]][["data_distribution"]]["Landings"]
+
+    landings_uncertainty <- get_data(data) |>
+      dplyr::filter(name == fleet_name, type %in% c("landings")) |>
+      dplyr::arrange(dplyr::desc(type)) |>
+      dplyr::pull(uncertainty)
+
+    landings_distribution_default <- switch(landings_distribution,
+                                         "DnormDistribution" = create_default_DnormDistribution(
+                                           value = landings_uncertainty,
+                                           input_type = "data",
+                                           data = data
+                                         ),
+                                         "DlnormDistribution" = create_default_DlnormDistribution(
+                                           value = landings_uncertainty,
+                                           input_type = "data",
+                                           data = data
+                                         )
+    )
+    names(landings_distribution_default) <- paste0(
+      landings_distribution,
+      ".",
+      names(landings_distribution_default)
+    )
 
-  index_distribution_default <- switch(index_distribution,
-    "DnormDistribution" = create_default_DnormDistribution(
-      value = index_uncertainty,
-      input_type = "data",
-      data = data
-    ),
-    "DlnormDistribution" = create_default_DlnormDistribution(
-      value = index_uncertainty,
-      input_type = "data",
-      data = data
+  } else {
+    log_Fmort_default <- list(
```

Comment:
I hope F in this situation is not used at all in mortality/survival calculations. Seems it could be inefficient.

Reply:
This F is used which is why the tiny value is defaulted. We can't use zero because of the log transformations but this way the use doesn't have to specify if the fleet is fishing or survey. We didn't have any other good ideas for how to set this up because we cannot have if forks based on parameter values or it breaks the autodiff chain.

Reply:
but I thought there is an "is_index" boolean somewhere?

Reply:
We got rid of that input so the fleet could use both. Andrea has a refactor for estimation that replaces booleans with text options so we could replace the landings_in_weight boolean with landings_units and have weight, number, or none options then skip the landings calculations if none is entered?